force-app/main/default/classes/NextStepBatch.cls

```diff
@@ -21,37 +21,39 @@
 public with sharing class NextStepBatch implements Database.Batchable<sObject>{
```

Comment:
Try to declare your batch as stateful o it remembers the records that were processed.   https://trailhead.salesforce.com/trailblazer-community/feed/0D54V00007T4BknSAF